src/repository/storage/postgres/orm/sequelize/note.ts

```diff
@@ -324,4 +345,80 @@ export default class NoteSequelizeStorage {
       },
     });
   };
+
+  /**
+   * Get all parent notes of a note that a user has access to,
+   * by checking the team access.
+   * @param noteId - the ID of the note.
+   * @param userId - the ID of the user.
+   */
+  public async getAllNoteParents(noteId: NoteInternalId, userId: number): Promise<NotePublic[]> {
+    if (!this.teamModel || !this.noteRelationModel) {
+      throw new Error(`${this.model !== null ? 'TeamStorage: Note relation model is not defined' : 'TeamStorage: Note model is not defined'}`);
+    }
+
+    const parentNotes: NotePublic[] = [];
+    let currentNoteId: NoteInternalId | null = noteId;
+    /**
+     * Store notes that user can not access, to check the inherited team if has access
+     */
+    let storeUnaccessibleNote: Note[] = [];
+
+    while (currentNoteId != null) {
+      const teamMember = await this.teamModel.findOne({
+        where: {
+          noteId: currentNoteId,
+          userId,
+        },
+        include: {
+          model: this.model,
+          as: this.model.tableName,
+          required: true,
+        },
+      });
+
+      if (teamMember && notEmpty(teamMember.notes)) {
+        if (storeUnaccessibleNote.length > 0) {
+          parentNotes.push(...storeUnaccessibleNote.map(note => ({
+            id: note.publicId,
+            content: note.content,
+            updatedAt: note.updatedAt,
+            createdAt: note.createdAt,
+            creatorId: note.creatorId,
+          })));
```

Comment:
it's better to return notes returned from the db, instead of creating the new structure

Reply:
it is not a task for repository level